src/applications/static-pages/pension-how-do-i-apply-widget/components/App/index.unit.spec.js

```diff
@@ -1,76 +1,92 @@
 import React from 'react';
 import sinon from 'sinon';
+import { Provider } from 'react-redux';
+import { render, fireEvent } from '@testing-library/react';
 import { expect } from 'chai';
-import { shallow } from 'enzyme';
+import { $ } from 'platform/forms-system/src/js/utilities/ui';
 import { toggleLoginModal as toggleLoginModalAction } from '@department-of-veterans-affairs/platform-site-wide/actions';
 import { App, mapDispatchToProps, mapStateToProps } from '.';
 
-describe.skip('Pension Widget <App>', () => {
-  let replaceStateSpy;
-
-  beforeEach(() => {
-    replaceStateSpy = sinon.spy(window.history, 'replaceState');
-  });
-
-  afterEach(() => {
-    replaceStateSpy.restore();
-  });
+const store = ({ pensionFormEnabled = false } = {}) => ({
+  getState: () => ({
+    user: {
+      profile: {
+        loading: false,
+        savedForms: [],
+      },
+      login: {
+        currentlyLoggedIn: true,
+      },
+    },
+    featureToggles: {
+      loading: false,
+      // eslint-disable-next-line camelcase
+      pension_form_enabled: pensionFormEnabled,
+    },
+  }),
+  subscribe: () => {},
+  dispatch: () => {},
+});
 
+describe('Pension Widget <App>', () => {
   it('renders the pension widget app', () => {
-    const wrapper = shallow(<App />);
-    expect(wrapper.find('h3').text()).to.equal(
+    const mockStore = store();
+    const { container } = render(
+      <Provider store={mockStore}>
+        <App />
+      </Provider>,
+    );
```

Comment:
Why do the new tests need:
* `render` vs `shallow`
* `<Provider>`
  * and thus a `store`

?

Reply:
I switched over to React Testing Library (`render`) from Enzyme (`shallow`) because enzyme does not render what's expected on the screen and it's easier to test. The Provider is required because the `useFeatureToggles` internals use a `useSelector` and not passed down by props.